pkg/time/time.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+// package time is a wrapper for the upstream system library for time.Time that
+// aliases most underlying types, but allows overrides for testing purposes.
+//
+// Synced to go-1.20.7.
+package time
```

Comment:
we can add the check similar to lock-check.sh if this new time package is supposed to be used in most of the cases.